### PR TITLE
EIP1-5755: Service impl to search AEDs by surname

### PR DESCRIPTION
**Pull request by vishalgupta-valtech**

### CODE REVIEW

File: src/test/kotlin/uk/gov/dluhc/printapi/service/aed/SearchDataSanitizationTest.kt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.CsvSource
+
+internal class SearchDataSanitizationTest {
+
+    @CsvSource(
+        value = [
+            "doe,DOE",
```
Comment (NathanRussellValtech) [inline src/test/kotlin/uk/gov/dluhc/printapi/service/aed/SearchDataSanitizationTest.kt:11]: Minor - do we need all these tests, and do we need so many examples for each "type" of sanitization?
Would 1 parameterized test like this do the same thing?
```
@CsvSource(
    value = [
        "doe,DOE",
        "  doe  ,DOE",
        "o'leary,OLEARY",
        "Llewelyn-Bowen,LLEWELYN BOWEN",
        "Harper  Adams,HARPER ADAMS"
    ]
)
```

Comment (vishalgupta-valtech) [reply src/test/kotlin/uk/gov/dluhc/printapi/service/aed/SearchDataSanitizationTest.kt:11]: It's a copy paste from vca, so didn't really review these tests

Comment (vishalgupta-valtech) [reply src/test/kotlin/uk/gov/dluhc/printapi/service/aed/SearchDataSanitizationTest.kt:11]: I think it's just testing the various aspects of sanitization but can merge them if you want

File: src/main/kotlin/uk/gov/dluhc/printapi/service/aed/AnonymousElectorDocumentSearchService.kt [hunk 2]
```diff
@@ -23,16 +26,36 @@ class AnonymousElectorDocumentSearchService(
...
         with(dto) {
-            val gssCodes = eroService.lookupGssCodesForEro(eroId)
-            val pagedSummaries = anonymousElectorDocumentSummaryRepository
-                .findAllByGssCodeInAndSourceType(
-                    gssCodes = gssCodes,
-                    sourceType = ANONYMOUS_ELECTOR_DOCUMENT,
-                    pageRequest = buildPageRequest(page = page, pageSize = pageSize)
-                ).map { anonymousSearchSummaryMapper.toAnonymousSearchSummaryDto(it) }
+            val pageRequest = buildPageRequest(page = page, pageSize = pageSize)
+            return when (searchBy) {
```
Comment (NathanRussellValtech) [inline src/main/kotlin/uk/gov/dluhc/printapi/service/aed/AnonymousElectorDocumentSearchService.kt:43]: Minor (and I'm not asking you to change this), but this is probably OK for now based on current requirements, but if we ever get asked to add more search by parameters this is going to get hard to manage.
Its OK for now, but if we get asked to add another searchBy parameter I think we'll probably want to refactor this, and do something similar to VCA in respect of specification builder etc

**Review by NathanRussellValtech — APPROVED** [review]
A couple of minor comments, but LGTM 👍

**Review by matt-wills-valtech — APPROVED** [review]

**Pull request merged**